VHAR-5104 Rajapinnan ja Integraation toteutus

sirkkalap opened the pull request:
- Mallina VHAR-4562, tosin integraation suunta vaihtuu

sirkkalap committed:
Integraation hahmottelua
- Mallina VHAR-4562, tosin integraation suunta vaihtuu

sirkkalap committed:
Tievelho nimiavaruus yhdistetty velho_komponentti.clj alle.
- Siirretty koodauskahvien katselmoinnin perusteella.

sirkkalap committed:
Lisätty protokolla ja implementaatio dummy haulle

sirkkalap committed:
Tievelho nimiavaruus yhdistetty velho_komponentti.clj alle.
- Poistettu vanhat komponentit

sirkkalap committed:
Kutsutaan toteutusta testissä. Ei vielä assertoida.

sirkkalap committed:
Valittu toteutuksen nimi
- Voi olla, että tulee tarkempaa jakoa myöhemmin varustetyyppien mukaan

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

# Conflicts:
#	src/clj/harja/palvelin/integraatiot/velho/velho_komponentti.clj
#	test/clj/harja/palvelin/integraatiot/velho/velho_komponentti_test.clj

sirkkalap committed:
Korjattu mergessä jäänyt nimeämisvirhe

sirkkalap committed:
Jaettu palvelu useampaan eri protocol:aan

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Varusteille oma konffi

sirkkalap committed:
Poistettu ylimääräinen kutsudata ja alustus

sirkkalap committed:
Poimittu yhteisiä apureita omaan namespaceen

sirkkalap committed:
Integraatiolle uusi tyyppi

sirkkalap committed:
Poimittu yhteisiä apureita omaan namespaceen

sirkkalap committed:
Varusterekisterin mallia testiin

petar-jr committed:
Tekee hae-token reusable

sirkkalap committed:
Korjattu funktioiden kutsuja vastaamaan aritya
- Toistaiseksi vain debug tulostusta

sirkkalap committed:
Ympäristön alustus ja HTTP metodi

sirkkalap committed:
SSL-engine käyttöön token-kutsussa

sirkkalap committed:
Korjattu parametrit vastaamaan uutta token-funktion signatuuria

sirkkalap committed:
Velho kohteiden haun toteutusta (oid-joukko)

sirkkalap committed:
Siistimistä

sirkkalap committed:
Korjattu boolean bugi

sirkkalap committed:
Ketjutettu rajapintakutsut

sirkkalap committed:
Muodostetaan oid-listan JSON oikein

sirkkalap committed:
Taulu uusille varustetoteumille

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Merge remote-tracking branch 'origin/VHAR-5104_Varustetietojen_perustiedot_HARJAsta' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Siivottu pois turha

sirkkalap committed:
Siistitty logiikkaa
- Tavoitteena nyt vain kaiteiden (TL501) haku
- Vastausten käsittelijät kesken
- Rajapinta URL kovakoodattu (asetukset.edn)
- Rajapinta aikarajaus kovakoodattu (asetukset.edn)
- Tietotyypin logiikka aloittamatta
- Kantaan kirjoitus aloittamatta

sirkkalap committed:
Tietolajin päättely funktio

sirkkalap committed:
Testidataa

sirkkalap committed:
Siivousta

sirkkalap committed:
Testitoteutusta TL päättelylle

sirkkalap committed:
Siistimistä

sirkkalap committed:
TL 505 Jäteastiat päättelyn toteutus

sirkkalap committed:
Siistimistä

sirkkalap committed:
Parempi rakenne testeille
- Koottu yhteisiä osia funktioiksi
- Lisätty TL505 testit

sirkkalap committed:
Muutettu käyttämään mappia

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_896__.sql

sirkkalap committed:
Mergen jälkeen siirretty migraatio eteenpäin

sirkkalap committed:
Lisätty testitiedostoon sisältö

sirkkalap committed:
TL506 päättely

sirkkalap committed:
Korjattu URL muistuttamaan paremmin oikeaa

sirkkalap committed:
Siistimistä

sirkkalap committed:
Ndjson tuki testiresursseihin

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_897__.sql

sirkkalap committed:
Mergen jälkeen siirretty migraatio eteenpäin

sirkkalap committed:
Katselmoinnin muutokset

sirkkalap committed:
Testitoteutus kohdelistan OID:ssa olevalle tietolajille

sirkkalap committed:
WC on tienvarsikalustetyyppi myös

sirkkalap committed:
Nostettu jäsennys funktioiksi

sirkkalap committed:
Lisätty virhekäsittelyn lokitus

sirkkalap committed:
Siistimistä

sirkkalap committed:
Korjattu muuttujanimi

sirkkalap committed:
Siivousta

sirkkalap committed:
Korjattu value-map truthy avainten poiminta

sirkkalap committed:
Korjattu tiedostolistaus (oli vahingossa rekursiivinen)

sirkkalap committed:
tyyppi=tvkt11 näyttäisi olevan riittävä ehto TL 504:lle

sirkkalap committed:
Siistimistä
- Lisätty loppuun rivinvaihdot

sirkkalap committed:
Lisää testidataa

sirkkalap committed:
Jaa tienvarsikalusteet omiin TL osiinsa

sirkkalap committed:
TL507 päättelyn toteutusTL507, 508 päättelyn toteutus

sirkkalap committed:
Varusterekisteristä haettu testidata

sirkkalap committed:
Poistettu Latauspalvelusta haettu testidata
- Testidata oli erimuotoista kuin Varusterekisteristä saatava.
- Puuttui kenttä `kohdeluokka`, jolloin testikoodiin jouduttiin kovakoodaamaan se.

sirkkalap committed:
Lisätty palvelu parametri

sirkkalap committed:
Siistimistä

sirkkalap committed:
TL509 toteutus

sirkkalap committed:
Noudettu kohteet uudelleen Varusterekisteristä
- Osa propertyistä puuttui Latauspalvelun kohteissa
- `kohdeluokka` propertyä tarvitaan, ettei sen arvoja tarvitse kovakoodata

sirkkalap committed:
Poistettu käyttämättömiä testitiedostoja

sirkkalap committed:
TL 509 testidata

sirkkalap committed:
Nimetty uudelleen testitiedosto

sirkkalap committed:
Tarkennuksia TL päättelyn nimikkeistöön

sirkkalap committed:
Muutettu päättely käyttämään vain kohteen tietoja

sirkkalap committed:
TL 512 toteutettu

- Muutettu päättely palauttamaan nil (ilman virhettä), jos se ei osaa päätellä tietolajia

sirkkalap committed:
Yhdistetty testidata ja siivottu pois tuplatestit

sirkkalap committed:
Pienennetty testijoukkoa

- Testattava osa ei ole riippuvainen datan eroista

sirkkalap committed:
Siivousta

sirkkalap committed:
TL512, TL513 toteutus

sirkkalap committed:
TL 515, 516 toteutus

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
TL-518, TL520

sirkkalap committed:
Nostettu http-kit versiota

sirkkalap committed:
Merge branch 'VHAR-4562-ota-velho-lahetys-kayttoon' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

# Conflicts:
#	src/clj/harja/palvelin/integraatiot/velho/velho_komponentti.clj

sirkkalap committed:
Vaihdettu migration numero konfliktin vuoksi

sirkkalap committed:
Palvelu debuggausta varten

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
TL522

sirkkalap committed:
Siivousta

sirkkalap committed:
Lukuvälien leikkaus

sirkkalap committed:
Merge branch 'VHAR-4562-ota-velho-lahetys-kayttoon' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

# Conflicts:
#	test/clj/harja/palvelin/integraatiot/velho/velho_komponentti_test.clj

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Siivousta

sirkkalap committed:
Siivousta

sirkkalap committed:
Urakka tiedon yhdistäminen varusteille

sirkkalap committed:
Siivousta

sirkkalap committed:
Poistetaan urakka_tr_sijainti taulu

- Tehdään sittenkin urakan haku urakan reunaviivageometrian perusteella

sirkkalap committed:
Poistetaan lukuvälien leikkaus -funktio

- Tehdään sittenkin urakan haku urakan reunaviivageometrian perusteella

sirkkalap committed:
Testi disabloitu, koska se on nyt rikki

sirkkalap committed:
Siivousta

sirkkalap committed:
Parannettu varustetoteuma2 constraineja

sirkkalap committed:
Poistetaan lukuvälien leikkaus -funktio

- Tehdään sittenkin urakan haku urakan reunaviivageometrian perusteella

sirkkalap committed:
Poistetaan lukuvälien leikkaus -funktio

- Tehdään sittenkin urakan haku urakan reunaviivageometrian perusteella

sirkkalap committed:
Lisää testejä urakkahaulle

sirkkalap committed:
Pois null testit. Not null sarakkeet.

sirkkalap committed:
Siivousta

sirkkalap committed:
TL522

sirkkalap committed:
Yhteinen toteutus varusteiden haulle eri palveluista

- Eri tietolajien lähteet listattu tietorakenteena `hae-varustetoteumat-velhosta` alussa.
- Haku looppaa nyt eri tietolajien lähteiden listan ja hakee eri palveluista muuttuneita varusteita
- Kohdetietojen haku käyttää nyt historia-tietojen endpointia, koska haluamme tietää, jos sama varuste on muuttunut
  monta kertaa haetulla aikavälillä
- Nostettu `hae-varustetoteumat-velhosta` ulos funktioita, jotta voidaan yksikkötestata niitä paremmin
- Kirjoitettu mahdollinen valmis toteutus runko haulle

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Korjauksia ja testejä varustehaulle

sirkkalap committed:
Parannettu virhekäsittelyä ja -viestejä

sirkkalap committed:
Korjauksia ja testejä varustehaulle

sirkkalap committed:
Siistitty nimeämistä

sirkkalap committed:
Siistitty nimeämistä

sirkkalap committed:
Siistitty nimeämistä

- Mietitty tarkemmin tunniste vs. oid nimeämistä

sirkkalap committed:
Korjattu nimeämisvirhe - väärä feature

sirkkalap committed:
Lisätty tietolajille lisätieto-kenttä

sirkkalap committed:
Poistettu kenttiä, joille ei ole tietolähdettä

sirkkalap committed:
Siivousta

sirkkalap committed:
Päivitetty nykyisen scheman mukaiseksi

sirkkalap committed:
Siirretty sanoma-pakettiin sinne kuuluvia osia

sirkkalap committed:
Testejä tietokantariveille

sirkkalap committed:
Varustetoteumien tallennus kantaan

sirkkalap committed:
Siistimistä

- Testien nimet päättyvät -test

sirkkalap committed:
Paljon muutoksia

- Pakollisten kenttien tarkistus
- Testejä konversiolle
- Virhetarkastelua ja tulosteita lisätty

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

# Conflicts:
#	test/clj/harja/palvelin/integraatiot/velho/velho_komponentti_test.clj

sirkkalap committed:
Siivousta

sirkkalap committed:
Kovennettu tl-testejä

sirkkalap committed:
Testi varustetoteuman idempotent toiminnalle

sirkkalap committed:
Testien tunniste resurssipolkuun

sirkkalap committed:
Siivousta

sirkkalap committed:
Siivousta

sirkkalap committed:
Ensin tarkempi poikkeus

sirkkalap committed:
Jatketaan, kun tallennus ei onnistu

sirkkalap committed:
Korjattu muokattu aikaleiman sanomamuodostus

sirkkalap committed:
Ylikirjoitetaan tiedot, jos sama varustetoteuma

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Dokumentoitu varusteen tietojen voimassa-olo ja historia tietoja testiin

sirkkalap committed:
Dokumentoitu varusteen tietojen voimassa-olo ja historia tietoja testiin

sirkkalap committed:
Muutettu vastaukset historia-endpointin mukaisiksi

sirkkalap committed:
Edellisen ajokerran muistaminen

sirkkalap committed:
Tallenna ja lokita varustetoteumahakuvirhe

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Revert "Muutettu vastaukset historia-endpointin mukaisiksi"

This reverts commit c3cbbe77c34a7765aceb1ef0677fa7e578efae86.

sirkkalap committed:
VHAR-5592 Varusteiden hakemisen testidata

sirkkalap committed:
Siivousta

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Konversioon liikennemerkkien otsikot

sirkkalap committed:
Siivousta

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Siivousta

sirkkalap committed:
Parannettu lokitusta

sirkkalap committed:
Konversio varustetoteuma kunto-ja-vauriotiedot yleinen-kuntoluokka

sirkkalap committed:
Uusin voittaa testi

sirkkalap committed:
Kuntoluokka tyyppi

sirkkalap committed:
Nimetty toteuma uudelleen

sirkkalap committed:
Toteuma

sirkkalap committed:
Siivousta

sirkkalap committed:
Validoinnin parannus

sirkkalap committed:
TL518 kivetyt alueet testi

sirkkalap committed:
TL514 melurakenteet - testi

sirkkalap committed:
Osajoukkohaun testi

sirkkalap committed:
Kohdeluokkien viimeksi-haettu aika

sirkkalap committed:
Osajoukkotestin korjaus

sirkkalap committed:
Korjattu integraatiotestien konfiguraatio

sirkkalap committed:
E2e testeille velho-varuste-salasana

sirkkalap committed:
Korjattu kohdeluokkien viimeksi haettu

sirkkalap committed:
Siivousta

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Siivousta

sirkkalap committed:
Debug tulostusta

sirkkalap committed:
Korjattu token url jalkeen parametri

sirkkalap committed:
Luodaan SSL-engine jokaiselle token pyynnölle erikseen

sirkkalap committed:
Siirrä pot-koodi omaan namespaceen

sirkkalap committed:
Siirrä varuste-toiminnot varuste ns-moduliin

sirkkalap committed:
Siirrä token haku yhteiset ns moduliin

sirkkalap committed:
Korjaa varuste-vastaanottosanoma testit

sirkkalap committed:
Toteuta cache velho-token haulle

sirkkalap committed:
Siirrä token haku jokaiseen api-kutsuun

sirkkalap committed:
Siivoa

sirkkalap committed:
Korjattu pois kaksinkertainen luiska lähde

sirkkalap committed:
Aseta kohdeluokkien viimeksi-haettu pvm.

sirkkalap committed:
Korjattu pois kaksinkertainen luiska lähde

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Poista vahingossa lisätty metodi

sirkkalap committed:
Siivoa

sirkkalap committed:
Korjaa Content-Type pyynnön bodylle

sirkkalap committed:
Muista virheet ettet päivitä viimeksi-haettu-pvm

sirkkalap committed:
Rikasta sijainnit kohteen pyynnöissä

sirkkalap committed:
Siivoa

sirkkalap committed:
Poista turhat DELETE lauseet testistä

sirkkalap committed:
Muuta aika Velhon tukemaan muotoon

sirkkalap committed:
Korjaa väärä suuruusvertailu tietolajien tunnistuksessa

sirkkalap committed:
Korjaa testin odotettu uuden toteutuksen mukaiseksi

sirkkalap committed:
Korjaa alkaen pvm Oulun MHU 2019-2024 aikavälille

sirkkalap committed:
Korjaa version tunniste

sirkkalap committed:
Käsittele json-jäsennyksen virheet

sirkkalap committed:
Käsittele sijaintipalvelun puuttuva historiatieto

sirkkalap committed:
Käsittele sijaintipalvelun puuttuva historiatieto

sirkkalap committed:
Poista ylimääräiset tietolajilähteet testiajosta

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Uusi testidata käyttöliittymän testausta varten

rckopteri commented on tietokanta/src/main/resources/db/migration/V1_998__petrisi.sql:
```diff
@@ -0,0 +1,392 @@
+-- Kaikkia tietoja ei haeta talteen Harjaan, vaan osa Varusteiden tiedoista saadaan jatkossakin
+-- kyselemällä tarvittaessa Velhon rajapinnasta.
+
+CREATE TABLE varustetoteuma2
```
Tämän voisi muuttaa vaikka varustetoteuma_velho nimiseksi. Tuo numero 2 viittaa enemmänkin testikoodiin ja väliaikaiseen tauluun.

sirkkalap replied:
Jep. Nimettiin taulu varustetoteuma_ulkoiset, joka kuvaa sitä, että toteumatiedot tulevat mahdollisesti useammasta ulkoisesta järjestelmästä. Samalla velho_oid muuttui ulkoinen_oid.

rckopteri commented on tietokanta/src/main/resources/db/migration/V1_998__petrisi.sql:
```diff
@@ -0,0 +1,392 @@
+    kuntoluokka      kuntoluokka_tyyppi    NOT NULL,
+    alkupvm          date                  NOT NULL,
+    loppupvm         date,
+    muokkaaja        text                  NOT NULL,
```
integer?

sirkkalap replied:
Ehkä pitää vaihtaa nimeksi velho_muokkaaja?

rckopteri replied:
Ah. joo. Jos tieto tulee Velhosta, niin hankala vaihtaa kolumnin tyyppiä. velho_muokkaaja olisi mielestäni parempi nimi.

sirkkalap replied:
Ai niin, mutta sitten TREX_muokkaaja tulisi, jos on TREX tietolähteenä?

sirkkalap replied:
Ja itseasiassa melkein kaikki tiedot ovat ulkoisia. Harja ei ole näiden master.

rckopteri commented on tietokanta/src/main/resources/db/migration/V1_998__petrisi.sql:
```diff
@@ -0,0 +1,392 @@
+(
+    id          serial PRIMARY KEY NOT NULL,
+    velho_oid   varchar(128)       NOT NULL,
+    alkupvm     timestamp          NOT NULL,
```
alku**pvm** viittaa päivämäärään, mutta tyyppinä timestamp. Ehdottaisin "alkuaika" termiä.

sirkkalap replied:
Se on Velho-domainin termi, mutta voidaan me laittaa siihen aika. Velho-domainissa se tarkoittaa Varusteen elinkaaren 1. päivää, joka ilmaistaan siellä jostain syystä kuitenkin aikaleimalla ja on nimetty päivämäärälsi. Sen effektiivinen vaikutus on Harjassa semmoinen, että jos se on yhtäsuuri version alku ajan kanssa, niin silloin on kyseessä create (luonti) toteumatyyppi.

sirkkalap replied:
Eikös tuo kenttä ole uudemmassa koodissa date? Mikäs on tuon commit id?

rckopteri commented on test/resurssit/velho/varusteet/varusterekisteri/README.md:
```diff
@@ -0,0 +1,7 @@
+Nämä tiedostot on muodostettu hakemalla Varusterekisteristä tai Varusterekisteristä kohteita, jotka kuuluvat
```
Varusterekisteristä
Varusterekisteristä

Mitä eroa on varusterekisterillä ja varusterekisterillä?

sirkkalap replied:
Hmm. Tässä on vaihtunut rekisteri refaktoroinnissa. :) Muut rekisterit ovat tulossa.

rckopteri commented on test/clj/harja/palvelin/main_test.clj:
```diff
@@ -176,7 +176,7 @@
     :materiaalit :selainvirhe :valitavoitteet :siltatarkastukset :lampotilat :maksuerat
     :liitteet :laadunseuranta :tarkastukset :ilmoitukset :tietyoilmoitukset
-    :turvallisuuspoikkeamat :integraatioloki-palvelu :raportit :yha :yha-velho :tr-haku
+    :turvallisuuspoikkeamat :integraatioloki-palvelu :raportit :yha :yha-velho :petrisi-velho :tr-haku
```
petrisi-velho -> velho?

sirkkalap replied:
Tässä on testikoodia. Merkattu, ettei vahingossa olis mergetty.

sirkkalap replied:
Tossa oli oikeasti kysymys bugista. Oli jäänyt vanha avain tuohon listaan.

rckopteri commented on src/clj/harja/palvelin/palvelut/varuste_velho.clj:
```diff
@@ -0,0 +1,54 @@
+                          (println "petrisi1419: kutsutaan hae-velho-varustetoteumat")
+                          (hae-velho-varustetoteumat velho user)))
+
+      ; TODO ***************************************
```
Vain muistutus, että nämä poistetaan

sirkkalap replied:
Joo, tätä käytetään testattaessa, koska muuten ei saa kokeiltua oikeaa Velhoa vastaan. Poistetaan ennen mergeä.

rckopteri commented on src/clj/harja/palvelin/palvelut/varuste_velho.clj:
```diff
@@ -0,0 +1,54 @@
+
+(defn hae-velho-varustetoteumat
+  [velho user]
+  ; TODO korja tämä kopioitu oikeustarkastus yha-velhosta
```
TODO

sirkkalap replied:
Jep. Oikeasti tätä ei mergetä.

rckopteri commented on src/clj/harja/palvelin/integraatiot/velho/varusteet.clj:
```diff
@@ -0,0 +1,385 @@
+(defn tuo-uudet-varustetoteumat-velhosta
+  [integraatioloki
+   db
+   {:keys [token-url
```
Jos tälle asettaisi vielä käyttämättömän parametrin :as asetukset, niin olisi helpompi lukea :)

sirkkalap replied:
Totta. Laitetaanpas.

rckopteri commented on src/clj/harja/palvelin/integraatiot/velho/varusteet.clj:
```diff
@@ -0,0 +1,385 @@
+                      :let (or (:etaisyys b) (:etaisyys a))}]
+      (:sijainti (first (q-toteumat/varustetoteuman-toimenpiteelle-sijainti db parametrit))))))
+
+(defn tuo-uudet-varustetoteumat-velhosta
```
Olisiko tämä funktio mahdollista pilkkoa? Aika monta sisäkkäistä try catch lausetta.

Esim funktio: tallenna-toteuma-fn joka luodaan letissä, voisi olla ihan erillinen funktio, jolloin sille voisi tehdä yksikkötestin.

Ja jotenkin hankalasti tuo tallennusfunktio välitetään parin funktion läpi johonkin missä sitä kutsutaan.
En osaa näin äkkikseltään sanoa, että voiko tuon tehdä eri tavalla, mutta näin katselmoidessa koodin lukeminen on hankalaa.

sirkkalap replied:
Olisiko tämä helpompi lukea?
![image](https://user-images.githubusercontent.com/703699/146939063-6215c4c9-3908-4a2a-adb4-5a1a502f9bd6.png)

rckopteri commented on README.md:
```diff
@@ -298,6 +298,11 @@ oikein eikä harja räsähdä.
 
+## Debug lokituksen näyttäminen
+
+Muokkaa asetukset.edn:aa jamuuta rivillä: 
```
Muokkaa asetukset.edn:aa ja muuta rivillä:

rckopteri commented on src/clj/harja/kyselyt/toteumat.sql:
```diff
@@ -787,6 +786,84 @@ WHERE id = :id;
 DELETE FROM varustetoteuma
 WHERE toteuma = :id;
 
+-- name: varustetoteuma-viimeisin-hakuaika-kohdeluokalle
```
Nimen perusteella hakee vain yhden, mutta haku hakee kaikki. Eikä order by parametrilla haeta viimeisintä. Vaan tässä haetaan kaikki, jonkun defaultin (ehkä id) perusteella.

Eli säätäisin joko hakua, tai haun nimeä

sirkkalap replied:
Näyttäisi olevan "outdated" ^^ Nykyinen koodi: ... `WHERE kohdeluokka = :kohdeluokka :: kohdeluokka_tyyppi;`

sirkkalap replied:
![image](https://user-images.githubusercontent.com/703699/146931644-8cff6a4b-fb74-4ffe-8341-e023b7a0db97.png)

rckopteri commented on src/clj/harja/kyselyt/toteumat.sql:
```diff
@@ -787,6 +786,84 @@ WHERE id = :id;
+
+-- name: luo-varustetoteuma2<!
+-- Luo uuden Velhosta tuodun varustetoteuman
+INSERT INTO varustetoteuma2 (velho_oid,
```
Tässä insertissä ei ole mitään erikoista, joten suosittelisin specql:n käyttöä tässä tilanteessa.

sirkkalap replied:
Jätetään näin, kun en tiedä paremmasta.

rckopteri commented on src/clj/harja/kyselyt/toteumat.sql:
```diff
@@ -787,6 +786,84 @@ WHERE id = :id;
+
+-- name: paivita-varustetoteuma2!
+-- Päivittää Velhosta tuodun varustetoteuman, joka oli jo kannassa. Uusin tieto voittaa!
+UPDATE varustetoteuma2
```
Tämänkin voisi tehdä specql:llä helpommin.

sirkkalap replied:
Lukija on hämmentynyt jälleen :)

rckopteri commented on src/clj/harja/kyselyt/toteumat.sql:
```diff
@@ -787,6 +786,84 @@ WHERE id = :id;
+
+-- name: tallenna-varustetoteuma2-kohdevirhe<!
+-- Tallentaa virheen tiedot tulevaa toipumista varten. Virheet tallennetaan velho-oid + muokattu avaimilla.
+INSERT INTO varustetoteuma2_kohdevirhe (velho_oid,
```
Ja taas specql:llää.. Sori kun toistan samaa näihin kaikkiin. Mutta specql on tosi hyvä yksinkertaisiin asioihin.

rckopteri commented:
Lisäsin useampia kommentteja ja tuosta ei oikein näe prioriteettiä niin avaan sitä hieman tähän.
Sanoisin, että se tallennuksen yksinkertaistaminen/luettavuuden parantaminen olisi ehkä se ykkösasia.
Sitten keskustelisin siitä tietokantataulujen nimeämisestä. En ole noiden numeroitujen taulujen fani.

Ja jos aikaa jää, niin vaihtaisin ne insertit/updatet käyttämään specql:ää.

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Nimeä uudelleen varustetoteuma2 taulu ja velho_oid

- Uudet nimet ovat varustetoteuma_ulkoiset ja ulkoinen_oid
- Testidatan nimi vaihdettu -> varustetoteumat_ulkoiset.sql
- Testidata tiedosto varustetoteumat_ulkoiset.sql includattu mukaan testidatojen syötössä

sirkkalap committed:
Lisää as asetukset, että on helpompi lukea

sirkkalap committed:
Typo

sirkkalap committed:
Poista turha sql-funktio

sirkkalap committed:
Jaa tallenna toiminnallisuus helpommin luettaviin osiin

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Siivousta

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Merge branch 'develop' into VHAR-5104_Varustetietojen_perustiedot_HARJAsta

sirkkalap committed:
Ignore idea-run kansio

sirkkalap committed:
Korjaa varustetoteuma-ulkoiset main osan nimi
